salt/_states/containerd.py

```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
             ret['result'] = True
         else:
             ret['comment'] = 'Failed to pull image'
+        return ret
+
+    return ret
```

Comment:
Maybe `return ret` only once?

Reply:
I really dislike the control flow of these dict-filling steps. Yes, you're absolutely right.